common/channel/many-keys-stress-test/src/main.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use channel::{libra_channel, message_queues::QueueStyle};
+use futures::{executor::block_on, stream::StreamExt};
+use std::{
+    io::{Cursor, Write},
+    num::NonZeroUsize,
+    sync::atomic::{AtomicBool, AtomicUsize, Ordering},
+    thread,
+    time::Duration,
+};
+use structopt::StructOpt;
+
+/// A small benchmark/stress test that sends `num_msgs` for each `num_keys`. The
+/// default arguments simulate many transient keys that just push a single message
+/// and then never more. Without garbage collecting empty per-key-queues, the
+/// program will eventually OOM.
+#[derive(Debug, StructOpt)]
+pub struct Args {
+    #[structopt(default_value = "2000000")]
+    num_keys: usize,
+
+    #[structopt(default_value = "1")]
+    num_msgs: usize,
+
+    #[structopt(default_value = "10")]
+    max_queue_size: usize,
+}
+
+fn main() {
+    let args = Args::from_args();
+    run(args);
+}
+
+pub fn run(args: Args) {
+    // Simulates an AccountAddress/PeerId
+    const KEY_SIZE: usize = 16; // bytes
+
+    // Simulates a (PeerManagerRequest, Option<Arc<_>>)
+    const MSG_SIZE: usize = 96; // bytes
```

Comment:
Personally, I find it useful to put sizing like `bytes` in the name of the constant, then you won't need the comment, and it'll be clear where you use it